setuptools/msvc.py

```diff
@@ -525,9 +528,24 @@ def VCInstallDir(self):
         """
         Microsoft Visual C++ directory.
         """
-        # Default path
-        default = r'Microsoft Visual Studio %0.1f\VC' % self.vc_ver
-        guess_vc = os.path.join(self.ProgramFilesx86, default)
+        self.VSInstallDir
```

Comment:
I appreciate your forward thinking in this block to give precedence to the future and indicate the past as legacy. Nice work.

One change I would make here is to implement these two techniques as a single assignment, like:

```
guess_vc = self._guess_vc() or self._guess_vc_legacy()
```